pkg/systemd/logs.js

```diff
@@ -175,17 +175,18 @@ $(function() {
         function load_service_filters(match, options) {
             loading_services = true;
             current_services = new Set();
-            const service_options = Object.assign({ output: "verbose" }, options);
-            let cmd = journal.build_cmd(match, service_options)[0].join(" ");
-            cmd += " | grep SYSLOG_IDENTIFIER= | sort -u";
-            cockpit.spawn(["sh", "-ec", cmd], { host: options.host, superuser: "try" })
-                    .then(function(entries) {
-                        entries.split("\n").forEach(function(entry) {
+            const service_options = Object.assign({ output: "cat" }, options);
+            const cmd = journal.build_cmd(match, service_options)[0];
+            cmd.splice(1, 0, "--output-fields=SYSLOG_IDENTIFIER");
```

Comment:
This is still a bit ugly -- how about adding that as a recognized `options` to `build_cmd()` itself? That already does a lot of similar things, e.g. for `!options.count` → `--no-tail`.

Reply:
Yeah, I was thinking about it, but I though it was too specific usecase to introduced it into `journal`. But sure, I can add it, it surely won't hurt anything and will be a bit cleaner.

Reply:
You can make it a bit more general, like passing arbitrary value for `output-fields` or so? then again, easy to adjust once we actually need it